### PR TITLE
Update Makefile and add docs to `cli/runtime` module

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it

- This is a followup PR of #3213 to fix Makefile under `cli/runtime` module and to add `docs` directory

### Which issue(s) this PR fixes

Related to #3213

### Describe testing done for PR
- Manually verified the Makefile changes

### Release note

```release-note
None
```

### CODE REVIEW

File: cli/runtime/hack/tools/Makefile [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+## --------------------------------------
+
+help: ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+## --------------------------------------
+## Binaries
+## --------------------------------------
+
+ginkgo: $(GINKGO)
```
Comment (marckhouzam) [inline cli/runtime/hack/tools/Makefile:40]: Do we want to keep this target?

File: cli/runtime/hack/tools/Makefile [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+$(GINKGO): go.mod
+	mkdir -p $(BIN_DIR)
+	go build -tags=tools -o $(BIN_DIR)/ginkgo github.com/onsi/ginkgo/ginkgo
+
+golangci-lint: $(GOLANGCI_LINT) $(SRCS) ## Build golangci-lint
```
Comment (marckhouzam) [inline cli/runtime/hack/tools/Makefile:45]: Do we want to keep this target?

File: cli/runtime/hack/tools/Makefile [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# Copyright 2021 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# Build tooling that is used for the main project
+
+# Ensure Make is run with bash shell as some syntax below is bash-specific
+SHELL := /usr/bin/env bash
+
+.DEFAULT_GOAL := help
```
Comment (marckhouzam) [inline cli/runtime/hack/tools/Makefile:9]: Cool, I didn't know about this.  No need to put the help target first anymore 😄

File: cli/runtime/hack/tools/Makefile [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+GOPROXY := $(shell go env GOPROXY)
+ifeq (,$(strip $(GOPROXY)))
+GOPROXY := https://proxy.golang.org
+endif
+export GOPROXY
+
+# Directories.
+ROOT_DIR:=$(shell git rev-parse --show-toplevel)
+BIN_DIR := bin
+TOOLS_SCRIPTS_DIR := $(ROOT_DIR)/hack/tools
```
Comment (marckhouzam) [inline cli/runtime/hack/tools/Makefile:21]: Not used (and mistakenly points to the root hack dir 😄 )

**Review by marckhouzam — COMMENTED** [review]
Good stuff.  Just some suggestions on the wording.

File: cli/runtime/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+- selector
+- question
+
+## Config Library
+
+This package implements helper functions to read, write and update the tanzu configuration file (`~/.config/tanzu/config.yaml`).
+
+## Plugin Helpers
+
+This package implements helper functions for new plugin creation. This is one of the main packages that each and every plugin will need to import to develop integrate with the Tanzu CLI.
```
Comment (marckhouzam) [inline cli/runtime/docs/README.md:26]: "develop integrate" : extra "develop"
That was my typo in my original comment. 😬

**Review by marckhouzam — APPROVED** [review]
Thanks for the quick changes.
One more that comes from my mistake.

LGTM after that.

File: cli/runtime/docs/README.md (hunk 5 above)
Comment (anujc25) [reply cli/runtime/docs/README.md:26]: My bad as well :)

**Review by marckhouzam — APPROVED** [review]

File: cli/runtime/docs/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+1. Component library
+1. Config library
+1. Plugin helpers
+1. Command helpers
+1. Test helpers
+
+## Component Library
+
+This package implements helper functions for the components listed below:
```
Comment (vuil) [inline cli/runtime/docs/README.md:13]: This package implements reusable CLI user interface components, including:

File: cli/runtime/docs/README.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
+# Tanzu CLI Integration Library
+
+This library exposes many different functionalities that plugin developers can use while developing Tanzu CLI plugins. Below are some of the functionalities that are available:
```
Comment (vuil) [inline cli/runtime/docs/README.md:3]: suggest:
This library provides functionality useful in the development of Tanzu CLI plugins. It includes:

File: cli/runtime/docs/README.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+This package implements helper functions to read, write and update the tanzu configuration file (`~/.config/tanzu/config.yaml`).
+
+## Plugin Helpers
+
+This package implements helper functions for new plugin creation. This is one of the main packages that each and every plugin will need to import to integrate with the Tanzu CLI.
+
+## Command Helpers
+
+This package implements command specific helper functions like command deprecation etc.
```
Comment (vuil) [inline cli/runtime/docs/README.md:30]:
```suggestion
This package implements command specific helper functions like command deprecation, etc.
```

**Review by vuil — APPROVED** [review]
lgtm, just some nits.

**Pull request merged**